misc/json-to-record-converter/README.md

```diff
@@ -0,0 +1,530 @@
+JSON
+```json
+{
+  "arrayProperty": [10,20,40]
+}
+```
+Ballerina
+```ballerina
+type NewRecord record {
+   int[] arrayProperty;
+};
+```
+This scenario can be applied for any type. No matter the type, if all values in the array are of the same type, then the record would get created with the field’s type as an array of the same type.
+
+#### Scenario 10: JSON array with mixed value types (oneOf)
+JSON
+```json
+{
+  "arrayProperty": [10, 20, 40, "AU", "UK"]
+}
+```
+Ballerina
+```ballerina
+type NewRecord record {
+   (int|string)[] arrayProperty;
+};
+```
+This is when there are more than one type of values present in the JSON array. The Ballerina record’s field type would be the union of the types of the values in the JSON array.
+
+#### Scenario 11: JSON array with same JSON objects
+JSON
+```json
+{
+  "arrayProperty": [{"name": "tom", "age": 5}, {"name": "jerry", "age": 4}]
+}
+```
+Ballerina
+```ballerina
+type ArrayPropertyItem record {
+   string name;
+   int age;
+};
+ 
+type NewRecord record {
+   ArrayPropertyItem[] arrayProperty;
+};
+```
+This scenario is when the JSON array only holds JSON objects and if all the objects schema is the same. If the Objects in the JSON array are different or if the array contains other types, then those are handled separately.
+
+#### Scenario 12: JSON array with different JSON objects
+In this case even if the objects in the array are completely different(the fields), still those objects will be treated the same with differencing fields having optional types (In such scenario all fields will become optional) See _Scenario 13_.
+
+JSON
+```json
+{
+  "arrayProperty": [{"name": "tom", "age": 5}, {"name": "jerry", "age": 4, "country": "SL"}]
+}
+```
+Ballerina
+```ballerina
+type ArrayPropertyItem record {
+   string name;
+   int age;
+   string country?;
+};
+ 
+type NewRecord record {
+   ArrayPropertyItem arrayProperty;
+};
+```
+
+#### Scenario 13: JSON array completely different JSON objects
+This is where all the fields are different in the objects in an array. (Even though the fields are different all these objects will be treated as same when converting to Ballerina record)
+
+JSON
+```json
+{
+  "arrayProperty": [{"name": "tom", "age": 5}, {"city": "Kandy", "zip": 71500}]
+}
+```
+Ballerina
+```ballerina
+type ArrayProperty record {
+   string name?;
+   int age?;
+   string city?;
+   int zip?;
+};
+ 
+type NewRecord record {
+   ArrayProperty[] arrayProperty;
+};
+```
+
+#### Scenario 14: JSON array with different types
+JSON
+```json
+{
+  "arrayProperty": [{"name": "tom", "age": 5}, "String Value", 5, 5.67]
+}
+```
+Ballerina
+```ballerina
+type ArrayPropertyItem record {
+   string name;
+   int age?;
+};
+ 
+type NewRecordList record {
+   (ArrayPropertyItem|decimal|int|string)[] arrayProperty;
+};
+```
+In this case, if there are more than two types of values in an array. This will be handles as union of data types in Ballerina record creation.
```

Comment:
```suggestion
This scenario has more than two types of values in an array. This will be handled as the union of data types in the Ballerina record creation.
```